e2e/flux_test.go

```diff
@@ -0,0 +1,28 @@
+package e2e
+
+import (
+	"testing"
+
+	"github.com/kluctl/kluctl/v2/e2e/test_resources"
+)
+
+func TestFluxCommands(t *testing.T) {
+	t.Parallel()
+
+	k := defaultKindCluster1
+
+	p := &testProject{}
+	p.init(t, k, "simple")
+
+	defer p.cleanup()
+
+	test_resources.ApplyYaml("flux.yaml", k)
+	test_resources.ApplyYaml("kluctl.yaml", k)
+	assertResourceExists(t, k, "flux-system", "kluctldeployment/microservices-demo-test")
+
+	// p.KluctlMust("flux-suspend", "--namespace", "flux-system", "--kluctl-deployment", "microservices-demo-test")
+
+	// p.KluctlMust("flux-resume", "--namespace", "flux-system", "--kluctl-deployment", "microservices-demo-test")
+	p.KluctlMust("flux-reconcile", "--namespace", "flux-system", "--kluctl-deployment", "microservices-demo-test", "--with-source")
```

Comment:
Hi @codablock So when running e2e tests I can't seem to figure out why kluctl command is failing as it only reports:
--- FAIL: TestFluxCommands (0.94s) project.go:439: Runnning kluctl: flux-reconcile --namespace flux-system --kluctl-deployment microservices-demo-test --with-source --no-update-check --debug project.go:470: project.go:471: kluctl failed: exit status 1

However when manually running the same command(against the same cluster-kind) it works without issues. Did I miss some param I need pass or ? Let me know, as this is the only think left to do I think.

Reply:
I can't say for sure tbh, but I have a few suggestions that might indirectly fix the issue that I have. I checked the new e2e test and it might be doing too much. It tries to install a fully fledged flux and flux-kluctl-controller installation which leads to multiple issues:
1. In kluctl.yaml, you have the `KluctlDeployment` listed BEFORE the CRD, which means that the test will always fail when run for the first time (and maybe for all further calls as well, unsure about that).
2. You added a ring-dependency. The Kluctl e2e code now depends on flux-kluctl-controller while the controller depends on the kluctl code ;)

I'd suggest to simplify the e2e test and not expect to actually install flux or any other controller. Change it to only install the KluctlDeployment CRD and the KluctlDeployment object. Then let the test only try to add the annotations and verify that they exist. I would suggest to not really test the controller functionality here, as this should be done as part of the controller e2e tests.

I'd also like to ask you to add a section in e2e/test_resources/README.md which gives instructions on how to re-create the test resources (e.g. when an upgrade is required). Easiest way is to add some bash lines to it, e.g:

```
# kluctl-crds.yaml
kustomize build https://github.com/kluctl/flux-kluctl-controller/config/crd > kluctl-crds.yaml

# kluctl-deployment.yaml
This is a simple KluctlDeployment that is not really valid. It points to a non-existing GitRepository. This object is only used to test `kluctl flux` subcommands (which only set annotations).
```

Reply:
So I've added `--no-wait` flag so it doesn't wait for it to get ready and I've removed deployment of controllers itself thus only deploying CRDs + example GitRepo + KluctlDeploy that needs to be annotated but still the same issue persists. What I'm starting to think is that he doesn't get cluster-config properly. Need to investigate this.

Reply:
Regarding your issue, did you do "go install ." before running the tests? Unfortunately, the e2e tests currently require you to put the compiled binary into a folder that is accessible through PATH. If you don't install the binary before running the tests, the tests will use the previously installed binary. I need to figure out a better way to ensure the latest compiled binary is used in e2e tests.

Reply:
Alright, so I did the merge as you suggested but I'm getting failure on other test now(sealed secret one). Didn't get around to see why exactly.